File: app/api/schemas/transaction.py
"""
Pydantic schemas for transaction data validation.
"""

from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional


class TransactionResponse(BaseModel):
    """Schema for transaction response."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    card_company: str
    amount: int
    transaction_date: datetime
    merchant: Optional[str] = None
    is_verified: bool = False


class MonthlySummaryCard(BaseModel):
    """Schema for per-card summary data."""

    card: str = Field(..., description="Card company name")
    total: int = Field(..., description="Total spending amount")
    count: int = Field(..., description="Number of transactions")


class MonthlySummaryResponse(BaseModel):
    """Schema for monthly summary response."""

    month: str = Field(..., description="Month in YYYY-MM format")
    by_card: list[MonthlySummaryCard] = Field(..., description="Summary by card company")
    grand_total: int = Field(..., description="Total spending across all cards")

File: src/types/transaction.ts
export interface CardTransaction {
  id?: number;
  card_company: string;
  amount: number;
  merchant: string;
  transaction_date: string;
  description: string;
  category: string | null;
  email_subject?: string;
  email_from?: string;
  gmail_message_id?: string;
  is_verified?: boolean;
  created_at?: string;
  memo?: string;
  tags?: string[];  // FlexTag: ["外食", "接待", "経費"]
}

export interface ParsedTransaction {
  amount: number;
  transaction_date: string; // ISO 8601: "YYYY-MM-DDTHH:MM:SS.sss..."
  merchant: string;
  card_company: string;
  raw_text: string;
}

export interface MonthlyAggregation {
  month: string;
  total: number;
  count: number;
  average: number;
  by_category: Record<string, number>;
  by_card: Record<string, number>;
}
